File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        res = ListNode()
        temp = res
        while list1 or list2:
            while list1 and (not list2 or list1.val <= list2.val):
                res.next = list1
                list1 = list1.next
                res = res.next
            while list2 and (not list1 or list2.val <= list1.val):
                res.next = list2
                list2 = list2.next
                res = res.next
        return temp.next

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    let temp = new ListNode();
    const res = temp;
    while (list1 || list2) {
        while (list1 && (!list2 || list1.val <= list2.val)) {
            temp.next = list1;
            list1 = list1.next;
            temp = temp.next;
        }
        while (list2 && (!list1 || list2.val <= list1.val)) {
            temp.next = list2;
            list2 = list2.next;
            temp = temp.next;
        }
    }
    return res.next;
};